test/unit/require.spec.ts

```diff
@@ -187,3 +187,165 @@ test("ImportEquals declaration require", () => {
+                export const value = defaultExport;
+            `,
+            "module.ts": `
+                export default 1 + 2 + 3;
+            `,
+        },
+        "value"
+    );
+
+    expect(result).toBe(6);
+});
+
+test("Import and Export Assignment", () => {
+    const [result] = util.transpileAndExecuteProjectReturningMainExport(
+        {
+            "main.ts": `
+                import * as m from "./module";
+                export const value = m;
+            `,
+            "module.ts": `
+                export = true;
+            `,
+        },
+        "value"
+    );
+
+    expect(result).toBe(true);
+});
+
+test("Mixed Exports, Default and Named Imports", () => {
+    const [result] = util.transpileAndExecuteProjectReturningMainExport(
+        {
+            "main.ts": `
+                import defaultExport, { a, b, c } from "./module";
+                export const value = defaultExport + b + c;
+            `,
+            "module.ts": `
+                export const a = 1;
+                export default a;
+                export const b = 2;
+                export const c = 3;
+            `,
+        },
+        "value"
+    );
+
+    expect(result).toBe(6);
+});
+
+test("Mixed Exports, Default and Namespace Import", () => {
+    const [result] = util.transpileAndExecuteProjectReturningMainExport(
+        {
+            "main.ts": `
+                import defaultExport, * as ns from "./module";
+                export const value = defaultExport + ns.b + ns.c;
+            `,
+            "module.ts": `
+                export const a = 1;
+                export default a;
+                export const b = 2;
+                export const c = 3;
+            `,
+        },
+        "value"
+    );
+
+    expect(result).toBe(6);
+});
+
+test("Export Default Function", () => {
+    const [result] = util.transpileAndExecuteProjectReturningMainExport(
+        {
+            "main.ts": `
+                import defaultExport from "./module";
+                export const value = defaultExport();
+            `,
+            "module.ts": `
+                export default function() {
+                    return true;
+                }
+            `,
+        },
+        "value"
+    );
+
+    expect(result).toBe(true);
+});
+
+test.each([
+    "export default class Test { static method() { return true; } }",
+    "export default class { static method() { return true; } }",
+])("Export Default Class", classDeclarationStatement => {
+    const [result] = util.transpileAndExecuteProjectReturningMainExport(
+        {
+            "main.ts": `
+                import defaultExport from "./module";
+                export const value = defaultExport.method();
+            `,
+            "module.ts": classDeclarationStatement,
+        },
+        "value"
+    );
+
+    expect(result).toBe(true);
+});
+
+test("Class exported by default has name `default`", () => {
```

Comment:
I think it also would be nice to have a test that ensures that named class retains it's specified name. Maybe reuse cases from the previous test or even merge them?